andela/ah-django
# 164859758 Implement push notifications

### What does this PR do?

Implement push notifications

### Description of Task to be completed?

- [x] Write tests for the endpoints for retrieving notifications
- [x] Set up signals for generating activity streams
- [x] Set up views and models for the activity streams

### How should this be manually tested?

1. Clone the repo to your local machine from [here](https://github.com/deferral/ah-django.git)
2. cd into the ah-django directory and checkout to the branch ```ft-system-notifications-164859758```

       cd ah-django

       git checkout ft-list-users-164859759

2. Create a virtual environment and activate it:

       python3 -m virtualenv venv && source venv/bin/activate

 * If you need to install virtualenv first:

       pip install virtualenv

4. Install the project dependencies

       pip install -r requirements.txt

5. Make database migrations and start the django server

       python3 manage.py makemigrations

       python3 manage.py migrate
 
       python3 manage.py runserver 

6. In your test client, register two new account by sending POST requests to ```http://127.0.0.1:8000/api/users```
7. Then use the given token to create an activities for each of the accounts. For example, use each of the accounts to follow the other., 
You should be able to see a profile with the username and email created with other fields set to null. 

### Any background context you want to provide?

More profiles are added as more accounts are created

### What are the relevant pivotal tracker stories?

[#164859759](https://www.pivotaltracker.com/story/show/164859759)
### Screenshots
![Screenshot from 2019-04-04 16-07-59](https://user-images.githubusercontent.com/38868860/55558052-e7f2c700-56f3-11e9-815a-a3139ffbf4e1.png)
![Screenshot from 2019-04-04 16-07-52](https://user-images.githubusercontent.com/38868860/55558053-e7f2c700-56f3-11e9-86b4-8c8111013524.png)
